core/Rename.cpp

```diff
@@ -122,20 +130,26 @@ namespace olympia
         if(dests.size() > 0)
         {
             sparta_assert(dests.size() == 1); // we should only have one destination
-            auto const & original_dest = inst_ptr->getRenameData().getOriginalDestination();
-            --reference_counter_[original_dest.rf][original_dest.val];
-            // free previous PRF mapping if no references from srcs, there should be a new dest mapping for the ARF -> PRF
-            // so we know it's free to be pushed to freelist if it has no other src references
-            if(reference_counter_[original_dest.rf][original_dest.val] <= 0){
-                freelist_[original_dest.rf].push(original_dest.val);
+            const auto dest = dests[0];
+            const auto rf  = olympia::coreutils::determineRegisterFile(dest);
+            const auto num = dest.field_value;
+            if (num != 0 || rf != core_types::RF_INTEGER)
```

Comment:
I think this would be easier to read if this:
```suggestion
            if (!is_x0) 
```

Reply:
Addressed in [650f6c6](https://github.com/riscv-software-src/riscv-perf-model/pull/120/commits/650f6c6b24edbaac804d3a6219ddc783a829b8cc)